### PR TITLE
Set Controller::render() arguments to view builder.

### PR DESCRIPTION
**Pull request by ADmad**
This allows checking template and layout values in beforeRender callbacks.

### CODE REVIEW

**Comment by saeideng** [comment]
now you can remove `$layout` arg from line 642 `$contents = $view->render($template, $layout);`

**Comment by ADmad** [comment]
Yeah but passing it as argument too doesn't hurt either.

**Comment by saeideng** [comment]
I want to say we can change view to use `$template` from the view obj instead of passing it again via `render()`

**Review by saeideng — APPROVED** [review]

File: src/Controller/Controller.php [hunk 1]
```diff
@@ -634,12 +642,12 @@ public function render(?string $template = null, ?string $layout = null): Respon
             return $this->response;
         }
 
-        if ($builder->getTemplate() === null && $this->request->getParam('action')) {
+        if ($builder->getTemplate() === null) {
```
Comment (markstory) [inline src/Controller/Controller.php]: Can `getParam('action')` ever return null?

Comment (ADmad) [reply src/Controller/Controller.php]: I don't think that's possible in normal execution but even if it is, you would be just setting existing null value to again null, no harm done.

**Pull request merged**